variable/include/scipp/variable/bin_variable.tcc

```diff
@@ -108,11 +108,20 @@ template <class T> class BinVariableMaker : public BinVariableMakerCommon<T> {
 private:
   const VariableConstView
   bin_parent(const scipp::span<const VariableConstView> &parents) const {
-    if (parents.empty())
+    constexpr auto is_bins = [](auto &x) {
+      return x.dtype() == dtype<bucket<T>>;
+    };
+    const auto count = std::count_if(parents.begin(), parents.end(), is_bins);
```

Comment:
Why count_if and not find_if? you can then store result and avoid calling `find_if` below?

Reply:
I would need to call `find_if` twice then to check if there is a second occurence? Would be much more complicated code?

Reply:
Yes I see also checking count > 1